File: server/src/main/java/by/brawl/ws/holder/AttributeHolder.java
package by.brawl.ws.holder;

public class AttributeHolder {
    private Integer max;
    private Integer current;

    public AttributeHolder(Integer bothValues) {
        max = bothValues;
        current = bothValues;
    }

    public AttributeHolder(Integer max, Integer current) {
        this.max = max;
        this.current = current;
    }

    public void incrementCurrent(Integer value) {
        current += value;
        if (current > max) {
            current = max;
        }
    }

    public void decrementCurrent(Integer value) {
        current -= value;
        if (current < 0) {
            current = 0;
        }
    }

    public Integer getMax() {
        return max;
    }

    public void setMax(Integer max) {
        this.max = max;
    }

    public Integer getCurrent() {
        return current;
    }

    public void setCurrent(Integer current) {
        this.current = current;
    }
}

File: server/src/main/java/by/brawl/ws/holder/AttributeHolder.kt
package by.brawl.ws.holder

class AttributeHolder {

    var max = 0
    var current = 0

    constructor(bothValues: Int) {
        max = bothValues
        current = bothValues
    }

    constructor(max: Int, current: Int) {
        this.max = max
        this.current = current
    }

    fun incrementCurrent(value: Int) {
        current += value
        if (current > max) {
            current = max
        }
    }

    fun decrementCurrent(value: Int) {
        current -= value
        if (current < 0) {
            current = 0
        }
    }

}